File: LeetCode/169.Majority_Element/169.Majority_Element.go
func majorityElement(nums []int) int {
    var res, count int

    for _, num := range nums {
        if(count == 0){
            res = num;
        }
        if num == res{
            count += 1
        }else{
            count -= 1
        }
    }

    return res
}

File: LeetCode/169.Majority_Element/169.Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {
        Integer candidate = null;
        int count = 0;

        for (int num : nums) {
            if (count == 0) {
                candidate = num;
            }

            if (candidate == num) {
                count += 1;
            } else {
                count -= 1;
            }
        }

        return candidate;
    }
}